optuna/integration/wandb.py

```diff
@@ -0,0 +1,144 @@
+    are logged and plotted as a function of optimizer steps.
+
+    .. note::
+        User needs to be logged in to Weights & Biases before
+        using this callback. For more information, please
+        refer to `wandb setup <https://docs.wandb.ai/quickstart#1-set-up-wandb>`_.
+
+    Example:
+
+        Add Weights & Biases callback to Optuna optimization.
+
+        .. code::
+
+            import optuna
+            from optuna.integration.wandb import WeightsAndBiasesCallback
+
+
+            def objective(trial):
+                x = trial.suggest_float("x", -10, 10)
+                return (x - 2) ** 2
+
+
+            n_trials = 10
+            wandbc = WeightsAndBiasesCallback(
+                n_trials=n_trials,
+                project_name="my-project-name",
+                group_name="my-group-name",
+                job_type="my-job-type",
+            )
+
+
+            study = optuna.create_study(study_name="my_study")
+            study.optimize(objective, n_trials=n_trials, callbacks=[wandbc])
+
+    Args:
+        n_trials:
+            The number of optimizer trials.
+        metric_name:
+            Name of the optimized metric. Since the metric itself is just a number,
+            `metric_name` can be used to give it a name. So you know later
+            if it was roc-auc or accuracy.
+        project_name:
+            Name of the project under which run should be categorized.
+            If the project is not specified, the run is put in an "Uncategorized" project.
+        group_name:
+            Specifies a group under project into which run should be categorized.
+            Please refer to `Group Runs
+            <https://docs.wandb.ai/guides/track/advanced/grouping>`_ for more details.
+        entity_name:
+            Username or team name under which run should be logged. When set to `None`,
+            the run is logged under the current user.
+        run_name:
+            Specifies the display name in Weights & Biases UI for this run.
+            When set to `None`, random two word name is generated instead.
+        job_type:
+            Specifies type of the run, and is used as additional grouping
+            for the runs.
+    """
+
+    def __init__(
+        self,
+        n_trials: int,
+        metric_name: Union[str] = "value",
+        project_name: Optional[str] = None,
+        group_name: Optional[str] = None,
+        entity_name: Optional[str] = None,
+        run_name: Optional[str] = None,
+        job_type: Optional[str] = None,
+    ) -> None:
+
+        _imports.check()
+
+        self._n_trials = n_trials
+        self._metric_name = metric_name
+        self._project_name = project_name
+        self._group_name = group_name
+        self._entity_name = entity_name
+        self._run_name = run_name
+        self._job_type = job_type
+
+    def __call__(self, study: optuna.study.Study, trial: optuna.trial.FrozenTrial) -> None:
```

Comment:
How about storing the same attributes as in [mlflow's callback](https://github.com/optuna/optuna/blob/cbfefe350d38aa47afff49094be0da1a60ee3d3f/optuna/integration/mlflow.py#L128-L141);
precisely, `distributions`, `direction` and `status`?

Reply:
Yes, I think logging trial related attributes could be solved with [a table with ](https://docs.wandb.ai/guides/data-vis/tables), and trial related ones can go to config as it is done now. Unfortunately tagging does not work in the same way as in `mlflow`.

Let me work on that some more.